### ISSUE TITLE
Build a simple GNN for dictionary creation

### ISSUE THREAD

**Issue by janetzki** [janetzki/dictionary_creator#22]
![image](https://user-images.githubusercontent.com/12159412/217789873-321fc0c1-1de9-4a8e-99d1-196bf4d0aa03.png)
(https://towardsdatascience.com/graph-neural-networks-with-pyg-on-node-classification-link-prediction-and-anomaly-detection-14aa38fe1275)
![image](https://user-images.githubusercontent.com/12159412/217307552-e119ef90-4c2f-48fc-9f3b-ba0486ab0f59.png)

## Goal
There is a GNN model that links LRL words with semantic domains.
Motivation: build GNN -> increase DC precision and recall > 0.30

## Tasks
- [x] #18  

## Notes
- Some ideas to get started
  - Re-implement https://medium.com/p/c71f6f61ff0f
    - https://github.com/plkmo/Bible_Text_GCN
    - use text-GCN for document classification for each of the 1189 chapters of the Bible
    - 66 labels = 66 books
    - each document-word and word-word edges has predetermined weights
      - eg tf-idf
    - chapter-word edges encode context within chapter
    - word-word edges encode relative context between chapters
    - https://miro.medium.com/max/1400/1*R2IFb9Q8HhtVlyxUa_cy3g.png
  - GNN: https://github.com/danielegrattarola/spektral
  - GNN: https://github.com/benedekrozemberczki/SimGNN
  - Try out https://github.com/pyg-team/pytorch_geometric
  - google "keras graph neural network (colab/notebook)"
  - if needed: get inspiration at https://paperswithcode.com/search?q_meta=&q_type=&q=graph+neural+networks+keras
